Android/app/src/androidTest/java/tie/hackathon/travelguide/Screens/MainPage.java

```diff
@@ -0,0 +1,25 @@
+package tie.hackathon.travelguide.Screens;
+
+import tie.hackathon.travelguide.R;
+import tie.hackathon.travelguide.Tests.Helpers;
+
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+
+/**
+ * Created by valentin.boca on 2/22/2018.
+ */
+
+public class MainPage {
+    public static void openNavDrawer() throws Exception {
+        Helpers.clickOnObjectWithContentDesc(R.string.navigation_drawer_open);
+    }
+
+    public static void signOut() throws Exception {
+        Helpers.clickOnObjectWithContentDesc(R.string.navigation_drawer_open);
+        Helpers.clickOnAView(R.id.nav_view, withId(R.id.design_navigation_view), 8);
+    }
+
+    public static boolean isScreenDisplayed(int rid) throws Exception{
```

Comment:
Hmm :-? screen-ul are un ID? Cand validam ca un screen este incarcat, verificam faptul ca cele mai reprezentative elemente (care fac diferenta dintre un screen si altul) de pe acel screen sunt displayed.

Reply:
Asa arata o metoda "waitFor":

    /**
     * Waits for CreditCard Screen
     */
    public static void waitFor() throws Exception {
        if (!isCreditCardDisplayed() && !isTitleDisplayed()) {
            throw new Exception("Not on Credit card screen.");
        }
    }

Reply:
Primele 2 metode dintr-un screen trebuie sa fie "waitFor" si "navigateTo". 💃